axes/backends.py

```diff
@@ -30,10 +31,13 @@ def authenticate(self, request, username=None, password=None, **kwargs):
         :return: Nothing, but will update return_context with lockout message if user is locked out.
         """
 
+        # Create credentials dictionary from username field
+        credentials = {settings.AXES_USERNAME_FORM_FIELD: username}
```

Comment:
lets pass all the authenticate params here including extra params. that would be useful and consistently formated for people using their own callables. WE don't need the "settings.AXES_USERNAME_FORM_FIELD" key as we already know the username will be in the 'username' param. (make sure to update the usage in def get_client_username to match too)

Reply:
Fine with the first remark. but the second seems wrong.
We know the username will be coming from the username function argument, but we don't know if the custom parsing function or the custom fieldname for all the other ways an authentication could be logged in AXES is expected to be username.
The get_client_username function is used in many places and completely oblivious to where it got called from.

We know Django Admin always uses username as the field and the default authentication backends do the same.
But what if someone uses the default Django Admin and a custom JSON supplied authentication in the same project?
The get_cleint_username would have to check both the credentials['username'] from Django Admin and the credentials[settings.AXES_USERNAME_FORM_FIELD] in such a case.

Reply:
There should probably be a `get_credentials` method that can be overridden here.

`username` argument is not always supplied here, actually. The argument might be `token` or something similar as well, it's up to the user. This signature just comes from the `ModelBackend` base class, but the Django authentication framework just specifies that the signature is `authenticate(request=None, **credentials)`.